src/analysis/retail/deathknight/unholy/modules/talents/CommanderOfTheDead.tsx

```diff
@@ -0,0 +1,121 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, SummonEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { formatPercentage } from 'common/format';
+
+class CommanderOfTheDead extends Analyzer {
+  commanderBuffs = 0;
+  petSummons = 0;
+  petSummonIDs = [
+    SPELLS.MAGUS_SUMMON.id,
+    SPELLS.APOC_SUMMON.id,
+    SPELLS.ARMY_SUMMON.id,
+    TALENTS.SUMMON_GARGOYLE_TALENT.id,
+    SPELLS.DARK_ARBITER_TALENT_GLYPH.id,
+  ];
+  buffedPets: string[] = [];
+  summonedPets: string[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.COMMANDER_OF_THE_DEAD_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.COMMANDER_OF_THE_DEAD_BUFF),
+      this.onBuffEvent,
+    );
+
+    this.addEventListener(Events.summon.by(SELECTED_PLAYER), this.onSummonEvent);
+  }
+
+  onBuffEvent(event: ApplyBuffEvent) {
+    const summonId = event.targetInstance
+      ? event.targetID.toString() + '|' + event.targetInstance.toString()
+      : event.targetID.toString(); // This is needed since the buff sometimes applies twice to the same summon.
+    if (!this.summonedPets.includes(summonId)) {
+      // This is the rare case of a pet being summoned without a summon event (potentially pre-combat).
+      this.summonedPets.push(summonId);
+      this.petSummons += 1;
+    }
+    if (!this.buffedPets.includes(summonId)) {
+      // Account for the case of double-buffing the same mob.
+      this.commanderBuffs += 1;
+      this.buffedPets.push(summonId);
+    }
+  }
+
+  onSummonEvent(event: SummonEvent) {
+    if (this.petSummonIDs.includes(event.ability.guid)) {
+      // We keep track of what has been summoned in case of a buff event on a minion that doesn't have a proper summon event.
+      const summonId = event.targetInstance
```

Comment:
We have helpers to better make ids for these things. Look at the usage of encodeEventSourceString and encodeEventTargetString in something like DireCommand

Reply:
Working on this, just need to sort out the fact that the encode function can return null type